src/api/operations/create_prediction.py

```diff
@@ -0,0 +1,53 @@
+"""
+Provides the API endpoint that makes a prediction on a target variable within a given set of data.
+"""
+
+import os
+
+from dotenv import load_dotenv
+from fastapi import APIRouter, HTTPException, Security
+from fastapi.security import APIKeyHeader
+from fastapi.encoders import jsonable_encoder
+from fastapi.responses import JSONResponse
+
+from api.resources.airport import Airport
+from api.resources.post_air_quality_response import PostAirQualityResponse
+from src.utils.logging import setup_logger
+
+# import logger
+logger = setup_logger()
+
+# import API key from environment variables
+load_dotenv()
+API_KEY = os.getenv("API_KEY")
+api_key_header = APIKeyHeader(name="create-prediction-API-key")
+
+# API key authentication method
+def get_api_key(api_key_attempt: str = Security(api_key_header)) -> str:
+    if api_key_attempt == API_KEY:
+        return api_key_attempt
+    raise HTTPException(
+        status_code=401,
+        detail="Unauthorised: Invalid or missing API key",
+    )
+
+prediction_router = APIRouter(prefix="/air-quality")
+
+@prediction_router.post("", response_model=PostAirQualityResponse, status_code=201)
+async def predict_air_quality(airport: Airport, api_key: str = Security(get_api_key)) -> JSONResponse:
+    logger.info("API key attempt: %s", api_key)
+
+    # Use airport input parameters to create air quality prediction and store as PostAirQualityResponse object
+    air_quality_response = PostAirQualityResponse(air_quality=get_air_quality_prediction(airport))
+
+    # Convert the object to a dictionary to allow it to be JSONified
+    response_dict = jsonable_encoder(air_quality_response)
```

Comment:
Is this needed? I'm not sure this couldn't be done in the `PostAirQualityResponse` model, or directly here?